client-app/src/Components/DonorForm.tsx

```diff
@@ -2,6 +2,294 @@ import React, { useState, ChangeEvent, FormEvent } from 'react';
 import axios from 'axios';
 import '../css/DonorForm.css';
 
+interface FormData {
+    itemType: string;
+    currentStatus: string;
+    donorEmail: string;
+    program: string;
+    imageUpload: string[];
+    dateDonated: string;
+}
+
+interface FormErrors {
+    [key: string]: string;
+}
+
+const DonorForm: React.FC = () => {
+    const [formData, setFormData] = useState<FormData>({
+        itemType: '',
+        currentStatus: 'Received',
+        donorEmail: '',
+        program: '',
+        imageUpload: [],
+        dateDonated: '',
+    });
+
+    const itemTypeOptions = [
+        {value: 'bicycle', label: 'Bicycle'},
+        {value: 'computer', label: 'Computer'},
+    ]
+
+    const donorEmailOptions = [
+        {value: 'email1', label: 'cooldude@gmail.com'},
+        {value: 'email2', label: 'cplusplushater@icloud.com'},
+        {value: 'email3', label: 'ISEBestBuilding@yahoo.com'},
+    ]
+
+    const programOptions = [
+        {value: 'youthProgram', label: 'Youth Program'},
+        {value: 'retailSales', label: 'Retail Sales'},
+        {value: 'recycle', label: 'Recycle'},
+        {value: 'earnABicycle', label: 'Earn-a-bicycle'},
+        {value: 'earnAComputer', label: 'Earn-a-computer'},
+    ]
+
+    const convertToBase64 = (file: File): Promise<string> => {
```

Comment:
Image conversion to base64 function looks fine as it converts the file to textual format.